### PR TITLE
Add support for server-provided assets

### PR DESCRIPTION
**Pull request by Bamcane**
This is a working draft.

This draft uses the design of the map-embedded assets in #7296 

### Why not to use a extra file?
As we can see, the DDNet maps is bigger and bigger since we can use HTTPS download maps.
Also, I said, this is a working draft, so if this problem is too serious, we can change to other ways.

But now, map-embedded assets is the best way.

Now this draft support to play the server-provided sounds that embedded in the map anytime, as long as server needs

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
Allowing the server to play map sounds sounds fine to me.

Could be used to annoy users.

Any other opinions?

**Comment by Bamcane** [comment]
> Allowing the server to play map sounds sounds fine to me.
> 
> Could be used to annoy users.
> 
> Any other opinions?

I'm trying to let server set the entity textures by embedding game.png-like images.

File: datasrc/network.py [hunk 1]
```diff
@@ -369,6 +369,10 @@
 		NetIntAny("m_Layer"),
 		NetIntAny("m_EntityClass"),
 	]),
+
+	NetEventEx("MapSoundWorld:Common", "map-sound-world@netobj.ddnet.org", [
+		NetIntAny("m_SoundId", 0),
```
Comment (Robyt3) [inline datasrc/network.py:374]: I would also add the sound volume to the net event and messages: `NetIntRange("m_Volume", 0, 255)`

(The volume is actually unused by `CSound` at the moment, I'll fix that separately)

**Review by Robyt3 — COMMENTED** [review]
Seems useful for modders. If server hosters want to annoy users they could already do that with map sounds on their own I guess.

> I'm trying to let server set the entity textures by embedding game.png-like images.

I think it's fine to limit the scope of this PR to making map sounds usable by the server.

**Comment by gerdoe-jr** [comment]
> Could be used to annoy users.

fng -> `/earrape`

**Comment by Bamcane** [comment]
Hm..So do i need to add option `m_Volume` into the event and the message?

**Comment by Bamcane** [comment]
I think this job didn't need to be in one pull request, so the first sound work could be join to the game.

**Pull request closed without merging**